src/azul/security/auth.py

```diff
@@ -0,0 +1,102 @@
+# See https://allspark.dev.data.humancellatlas.org/dcp-ops/docs/wikis/Security/Authentication%20and%20Authorization/Setting%20up%20DCP%20Auth
+from logging import getLogger
+from typing import Dict, Any
+from urllib.parse import urlencode
+
+from azul import config
+from azul.security.verifier import (verify,
+                                    InvalidTokenError,
+                                    NonDecodableTokenError)
+
+logger = getLogger(__name__)
+
+fusillade_callback_url = f'{config.service_endpoint()}/auth/callback'
+
+
+def get_fusillade_hostname():
+    # This can be moved to config.
+    deployment_stage = get_fusillade_stage()
+    fusillade_subdomain =f'auth.{deployment_stage}' if deployment_stage else 'auth'
+    return f'{fusillade_subdomain}.data.humancellatlas.org'
+
+
+def get_fusillade_stage() -> str:
```

Comment:
I don't like expressing this mapping from deployment stage to Fusillade stage as code. We use `deployments/*/environment` for expressing differences between deployments.